### ISSUE TITLE
Adding Conditions/Requirements (whichever fits best)

### ISSUE THREAD

**Issue by bvc0115** [PluginBugs/Issues-ItemsAdder#3304]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same suggestion was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if the feature is already implemented.
- [X] I already searched on the **Discord** server to know if anyone already has a solution for this or if the feature was already suggested.

### Discord tag (optional)

Anthyus#6029

### Describe the solution you'd like

Greetings,
I would like to ask for Conditions/Requirements when using Events to execute Actions more specifically, getting us more possibilities to use these properties.
As an example I will use the model replacement as I need for certain blocks:
I want a 'block' to change it's model once a player interacts with it, but I need to verify if item held in main hand is the actual block we are interacting with, exactly like candles, if you place a blue candle, you will see one blue candle, and if you right click again now you will see two candles together, of course, it has to be the same blue candle in order to change the model, otherwise it won't.
So I would like to achieve this with events, but I need a condition to verify player's holding item first, meaning:
events:
  placed_block:
    interact:
      condition/requirement:
        item_in_mainhand: customblock   (or held: true   | as using other event as requirement)
      set_block:
        block: customblock_phase_2
        target: RELATIVE
        decrement_amount: 1
        
Another example could be: (another way to set conditions)
events:
  attack:
    ifTargetType: player
    actions:
      target_potion_effect:
        effect: POISON
        time: 20
    ifTargetType: zombie
    actions:
      target_potion_effect:
        effect: INSTANT_DAMAGE
        strength: 2
    

### Is your feature request related to a problem?

Not really, it's more like a limitation

### Describe alternatives you've considered

I tried to use the "custom_variations" property but it seems work randomly.
I also tried to use blockstates as used in minecraft, but that would be more like creating a datapack.

### Additional context

To better understand what I need, just use a candle on minecraft and you will see how it works in game.
The other example was more like an idea, you could add more conditions so actions can be executed in a specific way. It's like using ifs, &&(AND) or ||(OR) when you code, but inside the yml file.

**Comment by LoneDev6** [comment]
Good idea, I have to think if there is a better way to do that or if the provided solution is clean enough
